libs/core/src/lib/overflow-layout/overflow-layout.component.ts

```diff
@@ -0,0 +1,408 @@
+        this._subscribeToRtl();
+    }
+
+    /** @hidden */
+    private _listenToChanges(): void {
+        this._subscription.add(
+            this._items.changes.subscribe(() => {
+                setTimeout(() => {
+                    this._fitVisibleItems();
+                });
+            })
+        );
+
+        this._listenToSizeChanges(this._elRef.nativeElement, this._itemsWrapper.nativeElement);
+    }
+
+    /** @hidden */
+    private _listenToSizeChanges(...elements: HTMLElement[]): void {
+        elements.forEach((element) =>
+            this._subscription.add(
+                resizeObservable(element)
+                    .pipe(
+                        skip(1),
+                        filter(() => this._listenToItemResize),
+                        distinctUntilChanged(),
+                        debounceTime(30)
+                    )
+                    .subscribe(() => {
+                        setTimeout(() => {
+                            this._fitVisibleItems();
+                        });
+                    })
+            )
+        );
+    }
+
+    /** @hidden */
+    private _fitVisibleItems(): void {
+        this._listenToItemResize = false;
+        this._allItems = this._items.toArray();
+        this._visibleItems.forEach((i) => (i.containerRef.hidden = false));
+        this._allItems.forEach((item, index) => {
+            item.hidden = false;
+            item.index = index;
+        });
+        this._cdr.detectChanges();
+        const containerWidth = this._elRef.nativeElement.getBoundingClientRect().width;
+        const itemsContainerWidth = this._itemsWrapper.nativeElement.getBoundingClientRect().width;
+
+        if (
+            containerWidth >= itemsContainerWidth &&
+            this._visibleItems.length <= this.maxVisibleItems &&
+            this._hiddenItems.length === 0
+        ) {
+            this._showMore = false;
+            this._cdr.detectChanges();
+            this._listenToItemResize = true;
+            return;
+        }
+        this._showMore = true;
+        let fittingElmCount = 0;
+        let fittingElmsWidth = 0;
+        let shouldHideItems = false;
+        this._cdr.detectChanges();
+
+        const showMoreContainerWidth = this._showMoreContainer.nativeElement.getBoundingClientRect().width;
+        let layoutWidth = this._layoutContainer.nativeElement.getBoundingClientRect().width;
+
+        // Try to find all forced visible items
+        const forcedItemsIndexes = this._getForcedItemsIndexes();
+
+        forcedItemsIndexes.forEach((itemIndex) => {
+            const container = this._visibleItems.get(itemIndex);
+            if (!container) {
+                return;
+            }
+            const elementSize = this._getElementWidth(container.elementRef.nativeElement);
+
+            layoutWidth -= elementSize;
+        });
+
+        if (layoutWidth < 0 && forcedItemsIndexes.length > 0) {
+            throw new Error(
```

Comment:
`console.warn` could be more polite in this case. broken layout is better than broken functionality. forced items may be being set in the app, not by the developer in the code. what do you think?

Reply:
Agree